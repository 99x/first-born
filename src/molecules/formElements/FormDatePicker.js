import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { DatePicker } from "../../atoms/DatePicker";
import { Text } from "../../atoms/Text";

export class FormDatePicker extends Component {

    render() {
        const { label, ...otherProps } = this.props;

        return (
            <View style={styles.container}>
                <Text size="sub_heading" color="rgba(33, 33, 33, 0.87)">{label}</Text>
                <DatePicker {...otherProps} />
            </View>
        )
    }
}

FormDatePicker.propTypes = {
    label: PropTypes.string.isRequired,
    ...DatePicker.propTypes
}

const styles = StyleSheet.create({
    container: {
        width: "100%"
    }
})