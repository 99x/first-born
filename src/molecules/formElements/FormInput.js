import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { Input } from "../../atoms/Input";
import { Text } from "../../atoms/Text";
import { commonColors } from "../../utils/color";

export class FormInput extends Component {
    render() {
        const { label, ...otherProps } = this.props;

        return (
            <View style={styles.container}>
                <Text size="sub_heading" color={commonColors.darkGrey}>
                    {label}
                </Text>
                <Input {...otherProps} />
            </View>
        );
    }
}

FormInput.propTypes = {
    label: PropTypes.string.isRequired,
    ...Input.propTypes
};

const styles = StyleSheet.create({
    container: {
        width: "100%"
    }
});
