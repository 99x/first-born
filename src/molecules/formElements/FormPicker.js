import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { Picker } from "../../atoms/Picker";
import { Text } from "../../atoms/Text";
import { commonColors } from "../../utils/color";

export class FormPicker extends Component {
    render() {
        const { label, children, data, ...otherProps } = this.props;

        return (
            <View style={styles.container}>
                <Text size="sub_heading" color={commonColors.darkGrey}>
                    {label}
                </Text>
                <Picker {...otherProps}>
                    {children
                        ? children
                        : data
                        ? data.map((dataElement, key) => (
                              <Picker.Item {...dataElement} key={key} />
                          ))
                        : null}
                </Picker>
            </View>
        );
    }
}

FormPicker.propTypes = {
    label: PropTypes.string.isRequired,
    date: PropTypes.array,
    ...Picker.propTypes
};

const styles = StyleSheet.create({
    container: {
        width: "100%"
    }
});
