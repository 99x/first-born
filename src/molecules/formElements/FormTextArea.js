import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { TextArea } from "../../atoms/TextArea";
import { Text } from "../../atoms/Text";
import { commonColors } from "../../utils/color";

export class FormTextArea extends Component {

    render() {
        const { label, ...otherProps } = this.props;

        return (
            <View style={styles.container}>
                <Text size="sub_heading" color={commonColors.darkGrey}>{label}</Text>
                <TextArea {...otherProps} />
            </View>
        )
    }
}

FormTextArea.propTypes = {
    label: PropTypes.string.isRequired,
    ...TextArea.propTypes
}

const styles = StyleSheet.create({
    container: {
        width: "100%"
    }
})