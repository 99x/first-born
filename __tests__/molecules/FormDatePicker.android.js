import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { FormDatePicker } from "../../src/molecules/formElements/FormDatePicker";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "android";
    return Platform;
});

it("renders regular FormDatePicker", () => {
    const tree = renderer
        .create(<FormDatePicker label="Date of Birth" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders colored FormDatePicker", () => {
    const tree = renderer
        .create(<FormDatePicker label="Date of Birth" color="red" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders FormDatePicker with placeholder", () => {
    const tree = renderer
        .create(
            <FormDatePicker label="Date of Birth" placeholder="placeholder" />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
