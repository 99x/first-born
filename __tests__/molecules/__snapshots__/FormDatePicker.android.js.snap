// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FormDatePicker with placeholder 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(33, 33, 33, 0.87)",
          "fontSize": 15,
          "fontWeight": "normal",
          "textAlign": "left",
        },
      ]
    }
  >
    Date of Birth
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgba(33, 33, 33, 0.5)",
        "borderRadius": 10,
        "borderWidth": 0.9,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginVertical": 10,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      placeholder
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "fontSize": 20,
          },
          Object {
            "paddingRight": 15,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders colored FormDatePicker 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(33, 33, 33, 0.87)",
          "fontSize": 15,
          "fontWeight": "normal",
          "textAlign": "left",
        },
      ]
    }
  >
    Date of Birth
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgba(33, 33, 33, 0.5)",
        "borderRadius": 10,
        "borderWidth": 0.9,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginVertical": 10,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Date
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "fontSize": 20,
          },
          Object {
            "paddingRight": 15,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders regular FormDatePicker 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(33, 33, 33, 0.87)",
          "fontSize": 15,
          "fontWeight": "normal",
          "textAlign": "left",
        },
      ]
    }
  >
    Date of Birth
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgba(33, 33, 33, 0.5)",
        "borderRadius": 10,
        "borderWidth": 0.9,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginVertical": 10,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Date
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "fontSize": 20,
          },
          Object {
            "paddingRight": 15,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
